plan/predicate_push_down.go

```diff
@@ -152,11 +152,83 @@ func (p *LogicalJoin) PredicatePushDown(predicates []expression.Expression) (ret
 			return nil, nil, errors.Trace(err2)
 		}
 	}
+	p.updateEQCond()
+	for _, eqCond := range p.EqualConditions {
+		p.LeftJoinKeys = append(p.LeftJoinKeys, eqCond.GetArgs()[0].(*expression.Column))
+		p.RightJoinKeys = append(p.RightJoinKeys, eqCond.GetArgs()[1].(*expression.Column))
+	}
 	p.mergeSchema()
 	p.buildKeyInfo()
 	return
 }
 
+func (p *LogicalJoin) updateEQCond() {
+	lChild, rChild := p.children[0], p.children[1]
+	var lKeys, rKeys []expression.Expression
+	for i := len(p.OtherConditions) - 1; i >= 0; i-- {
+		need2Remove := false
+		if eqCond, ok := p.OtherConditions[i].(*expression.ScalarFunction); ok && eqCond.FuncName.L == ast.EQ {
+			lExpr, rExpr := eqCond.GetArgs()[0], eqCond.GetArgs()[1]
+			if expression.ExprFromSchema(lExpr, lChild.Schema()) && expression.ExprFromSchema(rExpr, rChild.Schema()) {
+				lKeys = append(lKeys, lExpr)
+				rKeys = append(rKeys, rExpr)
+				need2Remove = true
+			} else if expression.ExprFromSchema(lExpr, rChild.Schema()) && expression.ExprFromSchema(rExpr, lChild.Schema()) {
+				lKeys = append(lKeys, rExpr)
+				rKeys = append(rKeys, lExpr)
+				need2Remove = true
+			}
+		}
+		if need2Remove {
+			p.OtherConditions = append(p.OtherConditions[:i], p.OtherConditions[i+1:]...)
+		}
+	}
+	if len(lKeys) > 0 {
+		lProj := p.createProj(0)
+		rProj := p.createProj(1)
```

Comment:
If left or right child is already a `Projection`, we can reuse it instead of creating one